content_server.py

```diff
@@ -0,0 +1,75 @@
+from core.config import Configuration
+from core.model import (
+    DataSource,
+    Identifier,
+)
+from core.opds_import import (
+    SimplifiedOPDSLookup,
+    OPDSImporter,
+)
+from core.coverage import (
+    CoverageProvider,
+    CoverageFailure,
+)
+
+class ContentServerException(Exception):
+    # Raised when the ContentServer can't connect or returns bad data
+    pass
+
+class ContentServerCoverageProvider(CoverageProvider):
+    """Checks the OA Content Server for Records"""
+
+    CONTENT_SERVER_RETURNED_ERROR = "OA Content Server returned error"
+    CONTENT_SERVER_RETURNED_WRONG_CONTENT_TYPE = "Content Server served \
+            unhandleable media type"
+
+    def __init__(self, _db):
+        self._db = _db
+        content_server_url = Configuration.integration_url(
+            Configuration.CONTENT_SERVER_INTEGRATION, required=True)
+        self.content_server = SimplifiedOPDSLookup(content_server_url)
+        self.importer = OPDSImporter(self._db, DataSource.OA_CONTENT_SERVER)
+        input_identifier_types = [Identifier.GUTENBERG_ID]
+        output_source = DataSource.OA_CONTENT_SERVER
+        super(ContentServerCoverageProvider, self).__init__(_db,
+                "OA Content Server Coverage Provider",
+                input_identifier_types, output_source, workset_size=10)
+
+    def process_item(self, identifier):
```

Comment:
Since this uses the lookup protocol you can do a batch of work at a time. Look at MetadataWranglerCoverageProvider in circulation/coverage.py. I bet you could even refactor a lot of code into a common subclass.